pkg/controller/kogitoinfra/kogitoinfra_controller.go

```diff
@@ -116,121 +92,28 @@ type ReconcileKogitoInfra struct {
 func (r *ReconcileKogitoInfra) Reconcile(request reconcile.Request) (result reconcile.Result, resultErr error) {
 	log.Infof("Reconciling KogitoInfra for %s in %s", request.Name, request.Namespace)
 
-	// Requires only one KogitoInfra instance in this namespace
-	instances := &appv1alpha1.KogitoInfraList{}
-	if err := kubernetes.ResourceC(r.client).ListWithNamespace(request.Namespace, instances); err != nil {
-		return reconcile.Result{}, err
-	}
-	if len(instances.Items) > 1 {
-		return reconcile.Result{RequeueAfter: time.Duration(5) * time.Minute},
-			fmt.Errorf("There's more than one KogitoInfra resource in this namespace, please delete one of them ")
-	}
-
 	// Fetch the KogitoInfra instance
-	instance := &appv1alpha1.KogitoInfra{}
-	if exists, err := kubernetes.ResourceC(r.client).FetchWithKey(request.NamespacedName, instance); err != nil {
-		return reconcile.Result{}, err
-	} else if !exists {
-		return reconcile.Result{}, nil
-	}
-
-	// watcher will be nil on test env
-	if controllerWatcher != nil && !controllerWatcher.AreAllObjectsWatched() {
-		if (instance.Spec.InstallKafka && !controllerWatcher.IsGroupWatched(kafkav1beta1.SchemeGroupVersion.Group)) ||
-			(instance.Spec.InstallInfinispan && !controllerWatcher.IsGroupWatched(infinispanv1.SchemeGroupVersion.Group)) ||
-			(instance.Spec.InstallKeycloak && !controllerWatcher.IsGroupWatched(keycloakv1alpha1.SchemeGroupVersion.Group)) {
-			// try to add them
```

Comment:
Not sure if we need to remove this check. This is important since if we add a Watch to a cluster that does not have the Infinispan, Kafka, Keycloak CRDs installed, a lot of errors will be seeing in the logs, might not even work. Maybe adding in the reconcilers you created?

Reply:
I had already included these check in their respective reconciler. 
https://github.com/vaibhavjainwiz/kogito-cloud-operator/blob/25ffe8183c9dfa325fa814790b8f41ab1c442834/pkg/controller/kogitoinfra/infinispan/infinispan_reconciler.go#L68
https://github.com/vaibhavjainwiz/kogito-cloud-operator/blob/25ffe8183c9dfa325fa814790b8f41ab1c442834/pkg/controller/kogitoinfra/kafka/kafka_reconciler.go#L64